### ISSUE TITLE
Support for MOI 1

### ISSUE THREAD

**Issue by lassepe** [osqp/OSQP.jl#106]
OSQP currently pins MOI to 0.10.3. A few days ago, [MOI as reached 1.0](https://github.com/jump-dev/MathOptInterface.jl/releases/tag/v1.0.0). It would be great to support MOI 1.0 within OSQP and adjust the compat bounds accordingly.

**Comment by odow** [comment]
Closed by https://github.com/osqp/OSQP.jl/pull/110

**Comment by odow** [comment]
Yeah maybe leave this open. There's actually a bunch of issues when I look through the code.

These should all be implemented and fixed:
https://github.com/osqp/OSQP.jl/blob/c2ff58bc26faa2e5a4318d5d666e19f9cdab8a61/test/MOI_wrapper.jl#L69-L93

Let me take a look through. I have some time, and it'd be good to fix up OSQP.jl properly for JuMP 1.0.

**Comment by imciner2** [comment]
If you could hold off doing any major updates to OSQP.jl for a while that would be appreciated. We are working on rebasing it to work with the updated API and library components that will be in v1 of the C library (hopefully being released soon), and have started those modifications in the `vb/develop1` branch on this repo. We are hoping to get everything to a point where we will be ready for v1 of the C library by the end of July, at which point we can merge the interface changes and release a v1 here as well.

**Comment by imciner2** [comment]
Alternately, if you want to target the branch we are working on the 1.0 version on, feel free to make more changes. But we don't have the jll it refers to pushed anywhere right now.

**Comment by odow** [comment]
> If you could hold off doing any major updates to OSQP.jl for a while that would be appreciated.

Okay, sure.
